src/OrchardCore.Modules/OrchardCore.DataProtection/Startup.cs

```diff
@@ -0,0 +1,45 @@
+using System.IO;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Modules;
+
+namespace OrchardCore.DataProtection
+{
+    public class Startup : StartupBase
+    {
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+        private readonly IOptions<ShellOptions> _shellOptions;
+        private readonly ShellSettings _shellSettings;
+
+        public Startup(
+            IDataProtectionProvider dataProtectionProvider,
+            IOptions<ShellOptions> shellOptions,
+            ShellSettings shellSettings)
+        {
+            _dataProtectionProvider = dataProtectionProvider;
+            _shellOptions = shellOptions;
+            _shellSettings = shellSettings;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            var directory = Directory.CreateDirectory(Path.Combine(
```

Comment:
@sebastienros PTAL.

Reply:
That's good, same code as in Media, so it will also be configurable in ENV to set a custom/shared folder.